ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/termsandconditions/v1/termsandconditions/clientlibs/runtime/js/termsandconditionsview.js

```diff
@@ -0,0 +1,126 @@
+/*******************************************************************************
+ * Copyright 2023 Adobe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+(function () {
+
+    class TermsAndConditions extends FormView.FormPanel {
+        static NS = FormView.Constants.NS;
+        static IS = "adaptiveFormTermsAndConditions";
+        static bemBlock = 'cmp-adaptiveform-termsandcondition';
+        static selectors = {
+            self: "[data-" + this.NS + '-is="' + this.IS + '"]',
+            label: `.${TermsAndConditions.bemBlock}__label`,
+            description: `.${TermsAndConditions.bemBlock}__longdescription`,
+            qm: `.${TermsAndConditions.bemBlock}__questionmark`,
+            tooltipDiv: `.${TermsAndConditions.bemBlock}__shortdescription`,
+            tncContentWrapper: `.${TermsAndConditions.bemBlock}__content-container`,
+            closePopupButton: `.${TermsAndConditions.bemBlock}__close-button`
+        };
+
+        constructor(params) {
+            super(params);
+            this.#handleScroll()
+        }
+
+        addChild(childView) {
+            super.addChild(childView);
+            if (childView.getModel()._jsonModel.fieldType === 'checkbox-group') {
+                this.#addLinkClickListener(childView);
+            }
+            if (childView.getModel()._jsonModel.fieldType === 'checkbox') {
+                this.#handlePopup(childView);
+            }
+        }
+
+        #addLinkClickListener(childView) {
+            childView.getWidgets().querySelectorAll('label').forEach((item) => {
+                const checkboxInput = item.querySelector('input');
+                item.querySelector('a').addEventListener('click', (e) => {
+                    checkboxInput.click();
+                })
+            })
+        }
+
+        #handleScroll() {
+            const intersection = this.element.querySelector('.cmp-adaptiveform-termsandcondition__text-intersect');
+            if (intersection) {
+                const self = this;
+                const onIntersection = ([{isIntersecting}]) => {
+                    if (isIntersecting) {
+                        const approvalCheckbox = self.getModel()._children.filter(child => child.fieldType === 'checkbox')[0];
+                        approvalCheckbox.dispatch(new Event('custom:scrollDone'));
```

Comment:
Once intersection is found, you also need to unobserve on the target element, else there would be a memory leak

Reply:
Done